apps/api/v2/src/modules/organizations/controllers/organizations-membership.controller.ts

```diff
@@ -0,0 +1,117 @@
+import { API_VERSIONS_VALUES } from "@/lib/api-versions";
+import { GetMembership } from "@/modules/auth/decorators/get-membership/get-membership.decorator";
+import { Roles } from "@/modules/auth/decorators/roles/roles.decorator";
+import { ApiAuthGuard } from "@/modules/auth/guards/api-auth/api-auth.guard";
+import { IsMembershipInOrg } from "@/modules/auth/guards/memberships/is-membership-in-org.guard";
+import { IsOrgGuard } from "@/modules/auth/guards/organizations/is-org.guard";
+import { RolesGuard } from "@/modules/auth/guards/roles/roles.guard";
+import { CreateOrgMembershipDto } from "@/modules/organizations/inputs/create-organization-membership.input";
+import { OrgMembershipOutputDto } from "@/modules/organizations/outputs/organization-membership.output";
+import { OrganizationsMembershipService } from "@/modules/organizations/services/organizations-membership.service";
+import {
+  Controller,
+  UseGuards,
+  Get,
+  Param,
+  ParseIntPipe,
+  Query,
+  Delete,
+  Patch,
+  Post,
+  Body,
+} from "@nestjs/common";
+import { ApiTags as DocsTags } from "@nestjs/swagger";
+import { plainToClass } from "class-transformer";
+
+import { SUCCESS_STATUS } from "@calcom/platform-constants";
+import { ApiResponse, SkipTakePagination } from "@calcom/platform-types";
+import { Membership } from "@calcom/prisma/client";
+
+@Controller({
+  path: "/v2/organizations/:orgId/memberships",
+  version: API_VERSIONS_VALUES,
+})
+@UseGuards(ApiAuthGuard, IsOrgGuard, RolesGuard)
+@DocsTags("Organizations Memberships")
+export class OrganizationsMembershipsController {
+  constructor(private organizationsMembershipService: OrganizationsMembershipService) {}
+
+  @Get()
+  @UseGuards()
+  @Roles("ORG_ADMIN")
+  async getAllMemberships(
+    @Param("orgId", ParseIntPipe) orgId: number,
+    @Query() queryParams: SkipTakePagination
+  ): Promise<ApiResponse<OrgMembershipOutputDto[]>> {
```

Comment:
You have defined return type of the routes as :
```
Promise<ApiResponse<OrgMembershipOutputDto[]>>
```

but this would not generate swagger response, because ApiResponse is just a type, and not a class-validator class. For example see `v2/src/ee/schedules/schedules_2024_04_15/controllers/schedules.controller.ts` - it returns `Promise<CreateScheduleOutput_2024_04_15>`, which looks like:

```
export class CreateScheduleOutput_2024_04_15 {
  @ApiProperty({ example: SUCCESS_STATUS, enum: [SUCCESS_STATUS, ERROR_STATUS] })
  @IsEnum([SUCCESS_STATUS, ERROR_STATUS])
  status!: typeof SUCCESS_STATUS | typeof ERROR_STATUS;

  @ApiProperty({
    type: ScheduleOutput,
  })
  @IsNotEmptyObject()
  @ValidateNested()
  @Type(() => ScheduleOutput)
  data!: ScheduleOutput;
}

``` 

this makes sure that the swagger doc generated has status and data keys.

Reply:
To check the swagger doc generated start the api and go to http://localhost:5555/docs

Reply:
I would avoid putting a suffix version except if we really need to differentiate with the base version